src/device/device.go

```diff
@@ -26,10 +26,12 @@ type Device interface {
 // This should be the primary method for creating model.Device(s).
 func FromNewDevice(newDeviceArgs model.NewDevice) model.Device {
 	newDevice := model.Device{
-		ID:        uuid.New(),
-		Host:      newDeviceArgs.Host,
-		Port:      newDeviceArgs.Port,
-		Endpoints: []model.Endpoint{},
+		ID:          uuid.New(),
+		Name:        *newDeviceArgs.Name,
+		Description: *newDeviceArgs.Description,
```

Comment:
I don't think Name or Description should be on this object. I feel like we should have a device profile object that has metadata about each device.

This is because the devices won't know their names or descriptions. They should remain as generic as possible. The mapping between device (unique MAC address) and DeviceProfile will only happen in deviceCommander.